backend/debug/available_regs.ml

```diff
@@ -291,13 +302,24 @@ let rec available_regs (instr : M.instruction) ~(avail_before : RAS.t) : RAS.t =
         if M.operation_can_raise op
         then augment_availability_at_raise (ok avail_across);
         let avail_after =
-          RD.Set.union
-            (RD.Set.without_debug_info (Reg.set_of_array instr.res))
-            avail_across
+          (* If a result register will never be named, we can forget about it
+             for the purposes of this analysis. *)
+          let res =
+            Reg.set_of_array instr.res
+            |> Reg.Set.inter all_regs_that_might_be_named
```

Comment:
```suggestion
            Reg.inter_set_arry all_regs_that_might_be_named instr.res
```

Reply:
Typo in the suggestion, will apply manually